pyigm/abssys/utils.py

```diff
@@ -303,3 +346,58 @@ def read_ion_file(ion_fil, components, lines=None, linelist=None, tol=0.05*u.AA)
     return table
 
 
+def sum_ionN(tbl1, tbl2):
+    """ Sum two ion column density tables
+
+    Parameters
+    ----------
+    tbl1 : Table
+    tbl2 : Table
+
+    Returns
+    -------
+    sum_tbl : Table
+
+    """
+    # Instantiate and use data form original as starting point
+    sum_tbl = tbl1.copy()
+
+    # Loop through other
+    for row2 in tbl2:
+        # New?
+        Zion = (row2['Z'], row2['ion'])
+        try:
+            row1 = sum_tbl[(sum_tbl['Z'] == Zion[0])&(sum_tbl['ion'] == Zion[1])]
+        except KeyError:
+            # Add in the new row
+            sum_tbl.add_row(row2)
+        else:
+            idx = np.where((sum_tbl['Z'] == Zion[0]) &
+                           (sum_tbl['ion'] == Zion[1]))[0][0]
+            # Clm
+            flagN, logN, siglogN = ltaa.sum_logN(row1, row2)
+            sum_tbl['logN'][idx] = logN
+            # Error
+            sum_tbl['sig_logN'][idx] = siglogN
+            # Flag
+            flags = [row1['flag_N'], row2['flag_N']]
+            if 2 in flags:   # At least one saturated
+                flag = 2
+            elif 1 in flags: # None saturated; at least one detection
+                flag = 1
```

Comment:
I guess I am quibbling with the fact that if one is detected and one is 
an upper limit, the sum gets set as a detection, while it should be a 
<=. But maybe I am not following.

Michele


On 03/01/2016 14:01, J. Xavier Prochaska wrote:
> In pyigm/abssys/utils.py
> <https://github.com/pyigm/pyigm/pull/5#discussion_r48693363>:
>
>> +            # Add in the new row
>> +            sum_tbl.add_row(row2)
>> +        else:
>> +            idx = np.where((sum_tbl['Z'] == Zion[0]) &
>> +                           (sum_tbl['ion'] == Zion[1]))[0][0]
>> +            # Clm
>> +            flagN, logN, siglogN = ltaa.sum_logN(row1, row2)
>> +            sum_tbl['logN'][idx] = logN
>> +            # Error
>> +            sum_tbl['sig_logN'][idx] = siglogN
>> +            # Flag
>> +            flags = [row1['flag_N'], row2['flag_N']]
>> +            if 2 in flags:   # At least one saturated
>> +                flag = 2
>> +            elif 1 in flags: # None saturated; at least one detection
>> +                flag = 1
>
> No. I think it is coded correctly.
>
> If either is saturated, the result is returned as a lower limit (flag = 2).
> ElseIf neither is saturated and at least one is detected, it is a value
> (flag = 1)
> Else it must be a pair of non-detections (upper limit).
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pyigm/pyigm/pull/5/files#r48693363>.Web Bug from
> https://github.com/notifications/beacon/AL4mfw1eHd6eJF35_9q4Fvn2ld5Nodccks5pWSFGgaJpZM4G4tqP.gif
>

-- 
Michele Fumagalli
Durham University
michele.fumagalli@durham.ac.uk
www.michelefumagalli.com